### PR TITLE
Fix for Numba

### PR DESCRIPTION
**Pull request by izaid**

### CODE REVIEW

File: dynd/nd/functional.pyx [hunk 1]
```diff
@@ -70,7 +70,7 @@ cdef public object _jit(object func, intptr_t nsrc, const _type *src_tp):
 
     # Compile the function with Numba
     func.compile(signature)
-    compile_res = func._compileinfos[signature]
+    compile_res = func.overloads[signature]
```
Comment (insertinterestingnamehere) [inline dynd/nd/functional.pyx]: This isn't enough on its own. With the latest numba there's another error at line 81 where there's no attribute `jit_codegen`.

**Comment by izaid** [comment]
Ping @pitrou and @seibert.

**Comment by insertinterestingnamehere** [comment]
I'm still looking in to the crash in our code after an error is raised here. It's one of those weird recursive destructor stack overflows that crop up occasionally when error handling goes awry.

File: dynd/nd/functional.pyx (hunk 1 above)
Comment (pitrou) [inline dynd/nd/functional.pyx]: This is renamed to `codegen()` simply.

**Pull request merged**